parla/tasks.py

```diff
@@ -447,6 +447,124 @@ def get_current_devices() -> List[Device]:
+    """
+    :param amount: The number of bytes reserved in the scheduler from tasks for persitent data. \
+      This exists, not as any kind of enforced limit on allocation, but rather to let the scheduler \
+      have an accurate measure of memory occupancy on the GPU beyond just memory that's used \
+      only during a task's execution. It can be specified as an integer representing the nubmer of \
+      bytes, an ndarray (cupy or numpy), or a list of ndarrays.
+    :param device: The device object where memory is to be reserved. \
+      This must be supplied if amount is an integer \
+      and may be supplied for an array. In the case of a list or other iterable it must \
+      be supplied if any element of the list is not an array. This may be a list of \
+      devices if amount is a list of array objects.
+    """
+    from . import cpu
+    if isinstance(amount, int):
+        memsize = amount
+    elif hasattr(amount, '__cuda_array_interface__'):
+        import cupy
+        # cupy has the info we need and views are cheap,
+        # so get a view then read off the size/device info.
+        view = cupy.array(amount)
+        memsize = view.nbytes
+        if device is None:
+            device = view.device
+    else:
+        # Check if "amount" supports the buffer protocol.
+        # if it does, we're reserving memory on the CPU
+        # unless the user says otherwise. If it does not,
+        # then assume it's a list of amount parameters
+        # that need to be handled individually.
+        amount_must_be_iterable = False
+        try:
+            view = memoryview(amount)
+        except TypeError:
+            amount_must_be_iterable = True
+        else:
+            memsize = view.nbytes
+            if device is None:
+                device = cpu(0)
+        # Not a cpu array, so try handling amount as
+        # an iterable of things that each need to be processed.
+        if amount_must_be_iterable:
+            try:
+                iter(amount)
+            except TypeError as exc:
+                raise ValueError("Persistent memory spec is not an integer, array, or iterable object") from exc
+            if device is None:
+                with ExitStack() as stack:
+                    for arr in amount:
+                        inner_must_be_iterable = False
+                        try:
+                            arr.__cuda_array_interface__
+                        except AttributeError as exc:
+                            inner_must_be_iterable = True
+                        else:
+                            stack.enter_context(reserve_persistent_memory(arr))
+                        if inner_must_be_iterable:
+                            try:
+                                iter(arr)
+                            except TypeError as exc:
+                                raise ValueError("Implicit location specification only supported for GPU arrays.") from exc
```

Comment:
Why? There is `parla.array.get_memory(a).device`, which should do what you need without any of this.

Reply:
Great point. I wasn't aware that interface existed.